src/test/emotion-prefix.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { replaceEmotionPrefix } from './emotion-prefix';
+
+describe('replaceEmotionPrefix', () => {
+  it('matches EUI component Emotion classes', () => {
+    expect(replaceEmotionPrefix('css-abc1234-euiComponent')).toBe(
+      'emotion-euiComponent'
+    );
+  });
+  it('matches EUI component Emotion classes with variants', () => {
+    expect(replaceEmotionPrefix('css-abc1234-euiComponent-primary-m')).toBe(
+      'emotion-euiComponent-primary-m'
+    );
+  });
+  it('matches chained EUI component Emotion classes', () => {
+    expect(
+      replaceEmotionPrefix('css-abc1234-euiComponentStyles-EuiComponent')
+    ).toBe('emotion-euiComponentStyles-EuiComponent');
+  });
```

Comment:
I'd like to see this test modified to confirm that `__` selectors/descendants are also captured
```suggestion
  it('matches chained EUI component Emotion classes', () => {
    expect(
      replaceEmotionPrefix('css-abc1234-euiComponentStyles-EuiComponent__child')
    ).toBe('emotion-euiComponentStyles-EuiComponent__child');
  });
```

Reply:
I'd also like to see another test confirming that `css-abc1234-EuiComponentStyles` is captured as well since we added a capture case for it 🤷